File: core/src/com/spacefist/entities/EnemyMine.java
package com.spacefist.entities;

import com.badlogic.gdx.math.Rectangle;
import com.badlogic.gdx.math.Vector2;
import com.spacefist.GameData;
import com.spacefist.components.NullInputComponent;
import com.spacefist.components.Physics;
import com.spacefist.components.Sound;
import com.spacefist.components.Sprite;
import org.jetbrains.annotations.NotNull;

public class EnemyMine extends Entity {
    public EnemyMine(@NotNull GameData gameData, @NotNull Vector2 position) {
        super(
            gameData,
            new Rectangle(
                (int) position.x,
                (int) position.y,
                gameData.getTextures().get("EnemyMine").getWidth(),
                gameData.getTextures().get("EnemyMine").getHeight()
            ),
            new Physics(),
            new NullInputComponent(),
            new Sprite(gameData.getTextures().get("EnemyMine")),
            new Sound(gameData.getSoundEffects().get("Explosion")),
            0
        );
    }

    public void hit() {
        ((Sound) getSound()).play();
    }
}

File: core/src/com/spacefist/entities/EnemyMine.kt
package com.spacefist.entities

import com.badlogic.gdx.math.Rectangle
import com.badlogic.gdx.math.Vector2
import com.spacefist.GameData
import com.spacefist.components.NullInputComponent
import com.spacefist.components.Physics
import com.spacefist.components.Sound
import com.spacefist.components.Sprite

class EnemyMine(gameData: GameData, position: Vector2) : Entity(gameData, Rectangle(
        position.x.toInt().toFloat(),
        position.y.toInt().toFloat(),
        gameData.textures["EnemyMine"]!!.getWidth().toFloat(),
        gameData.textures["EnemyMine"]!!.getHeight().toFloat()
), Physics(), NullInputComponent(), Sprite(gameData.textures["EnemyMine"]!!), Sound(gameData.soundEffects["Explosion"]!!), 0.0f) {

    fun hit() {
        (sound as Sound).play()
    }
}
